pkg/controllers/cronfederatedhpa/cronfederatedhpa_handler_test.go

```diff
@@ -0,0 +1,300 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cronfederatedhpa
+
+import (
+	"testing"
+	"time"
+
+	"github.com/go-co-op/gocron"
+	"github.com/stretchr/testify/assert"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	autoscalingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/autoscaling/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+func TestCronFHPAScaleTargetRefUpdates(t *testing.T) {
+	handler := NewCronHandler(fake.NewClientBuilder().Build(), record.NewFakeRecorder(100))
+
+	tests := []struct {
+		name           string
+		cronFHPAKey    string
+		initialTarget  autoscalingv2.CrossVersionObjectReference
+		updatedTarget  autoscalingv2.CrossVersionObjectReference
+		expectedUpdate bool
+	}{
+		{
+			name:        "New scale target",
+			cronFHPAKey: "default/test-cronhpa",
+			initialTarget: autoscalingv2.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				Name:       "test-deployment",
+				APIVersion: "apps/v1",
+			},
+			updatedTarget: autoscalingv2.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				Name:       "test-deployment",
+				APIVersion: "apps/v1",
+			},
+			expectedUpdate: false,
+		},
+		{
+			name:        "Same scale target",
+			cronFHPAKey: "default/test-cronhpa",
+			initialTarget: autoscalingv2.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				Name:       "test-deployment",
+				APIVersion: "apps/v1",
+			},
+			updatedTarget: autoscalingv2.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				Name:       "test-deployment",
+				APIVersion: "apps/v1",
+			},
+			expectedUpdate: false,
+		},
+		{
+			name:        "Different scale target",
+			cronFHPAKey: "default/test-cronhpa",
+			initialTarget: autoscalingv2.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				Name:       "test-deployment",
+				APIVersion: "apps/v1",
+			},
+			updatedTarget: autoscalingv2.CrossVersionObjectReference{
+				Kind:       "StatefulSet",
+				Name:       "test-statefulset",
+				APIVersion: "apps/v1",
+			},
+			expectedUpdate: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handler.CronFHPAScaleTargetRefUpdates(tt.cronFHPAKey, tt.initialTarget)
```

Comment:
Is there a possibility of conflict when using the same handler for different test cases?

Reply:
Yeah, what you said makes sense. I'll update the tests to ensure that each test case uses a fresh handler instance.

Reply:
but the handler map has cronFHPAKey as the keys

Reply:
what do u suggest @zhzhuang-zju ?

Reply:
> but the handler map has cronFHPAKey as the keys

sorry, I didn't get it. Could you explain what issue this might cause?

Reply:
I meant using different cronFHPAKey values for each test case, even with the same handler instance, should not lead to conflicts. What do u say?

Reply:
This does indeed avoid conflicts, but using different cronFHPAKeys also increases maintenance costs in the long run.

Reply:
Okay, i'll use a sepearate cronFHPAKey for new scale target and we'll create a new handler for each test case. Is is good?

Reply:
yeah~make sense to me

Reply:
Thanks!